src/batch/src/executor/bench_utils.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::catalog::{Field, Schema};
+use risingwave_common::types::DataType;
+
+use super::test_utils::{gen_data, MockExecutor};
+use super::BoxedExecutor;
+
+#[macro_export]
+macro_rules! bench_join {
+    // TODO: change the type of $join.
+    ($join:expr) => {
+        paste! {
+            fn [<bench_ $join:snake _internal>](c: &mut Criterion, with_conds: Vec<bool>, join_types: Vec<JoinType>) {
```

Comment:
The name `bench_join!` doesn't look good. Should be sth like `make_bench_join`.

And the macro doesn't seem to be necessary here. To generate functions, I guess higher order function seems to be enough? It might be sth like:

```rust
type BenchFn = impl Fn(&mut Criterion, ...)
fn make_bench_join(
  create_join: fn(...)->BoxedExecutor,
  bench_id: &str
) -> BenchFn {
  ...
}
```

Reply:
I'm trying to introduce a module under` batch/benches/`
I will use high-order function instead of macro.